### ISSUE TITLE
fails to find cups

### ISSUE THREAD

**Issue by rochelmeyer** [husisusi/officeonlin-install.sh#103]
Hi all, 

checking whether to enable CUPS support... yes
checking whether cups support is present... checking for cupsPrintFiles in -lcups... yes
checking cups/cups.h usability... no
checking cups/cups.h presence... yes
configure: WARNING: cups/cups.h: present but cannot be compiled
configure: WARNING: cups/cups.h:     check for missing prerequisite headers?
configure: WARNING: cups/cups.h: see the Autoconf documentation
configure: WARNING: cups/cups.h:     section "Present But Cannot Be Compiled"
checking for cups/cups.h... configure: WARNING: cups/cups.h: proceeding with the compiler's result
configure: error: Could not find CUPS. Install libcups2-dev or cups-devel.
no
Error running configure at ./autogen.sh line 281.

I have cups and libcups2-dev installed. Libcups2-dev is version 2.1.3-4 amd64

**Comment by rochelmeyer** [comment]
cloned another libreoffice from github and ran ./autogen.sh there: 

checking whether to enable CUPS support... yes
checking whether cups support is present... checking for cupsPrintFiles in -lcups... yes
checking cups/cups.h usability... yes
checking cups/cups.h presence... yes
checking for cups/cups.h... yes
checking for pkg-config... /usr/bin/pkg-config

So, could it be something with cloning libreoffice from git went wrong? I don't see how this should happen and i did not get nay warnings, but this looks a bit strange to me.

**Comment by Kassiematis** [comment]
I think cloning must have gone wrong. Compiling with --disable-cups can also be a solution.

**Comment by rochelmeyer** [comment]
Thanks, will try that. I tried --without-cups but that (of course) did not have any effect.

**Comment by aalaesar** [comment]
Hi.
is this issue still relevant ?

**Comment by rochelmeyer** [comment]
Yes, it is. --disable-cups did not do the job. 
Sorry for answering a bit late!

**Comment by aalaesar** [comment]
Do you still have the issue from scratch ?

**Comment by rochelmeyer** [comment]
Will try that tomorrow. No chance to do this today. Work :-/

**Comment by rochelmeyer** [comment]
it now passes beyond the CUPS-point (ignores cups), but fails with another error: 

checking whether /usr/bin/ccache g++ supports thread safe statics... yes
checking whether /usr/bin/ccache gcc supports -fvisibility=hidden... yes
checking whether /usr/bin/ccache gcc supports -mno-avx... yes
checking whether /usr/bin/ccache gcc supports -fstack-protector-strong... yes
checking whether /usr/bin/ccache gcc supports atomic functions... yes
checking whether /usr/bin/ccache gcc supports __attribute__((deprecated(message)))... no
checking whether /usr/bin/ccache g++ defines __base_class_type_info in cxxabi.h... no
checking whether /usr/bin/ccache g++ defines __class_type_info in cxxabi.h... no
checking whether /usr/bin/ccache g++ declares __cxa_allocate_exception in cxxabi.h... no
checking whether /usr/bin/ccache g++ defines __cxa_eh_globals in cxxabi.h... no
checking whether /usr/bin/ccache g++ defines __cxa_exceptions in cxxabi.h... no
checking whether /usr/bin/ccache g++ declares __cxa_get_globals in cxxabi.h... no
checking whether /usr/bin/ccache g++ declares __cxa_throw in cxxabi.h... no
checking whether /usr/bin/ccache g++ defines __si_class_type_info in cxxabi.h... no
checking whether /usr/bin/ccache g++ defines __vmi_class_type_info in cxxabi.h... no
checking whether STL uses __attribute__((warn_unused))... yes
checking whether /usr/bin/ccache g++ supports __attribute__((warn_unused))... no
checking What the C++ library is... configure: error: Could not figure out what C++ library this is
Error running configure at ./autogen.sh line 281.
thr@xxxxxxxxxxxx:/tmp/officeonlin-install.sh$ sudo apt install gcc g++ ccache
Reading package lists... Done
Building dependency tree       
Reading state information... Done
g++ is already the newest version (4:5.3.1-1ubuntu1).
gcc is already the newest version (4:5.3.1-1ubuntu1).
ccache is already the newest version (3.2.4-1).

**Comment by aalaesar** [comment]
Hello.
I cannot reproduce the problem.
Maybe this issue is related to a later fix regarding escaped carriage return.
You may want to check the scripts logs, specially ` preparation.log`, for errors.
or retry with the latest version.

**Comment by Brakelmann** [comment]
Hi,

facing the same issue today.
Using the officeonline-install.sh script leads to the following error:
...
checking whether to enable CUPS support... yes
checking whether cups support is present... checking for cupsPrintFiles in -lcups... yes
checking cups/cups.h usability... no
checking cups/cups.h presence... yes
configure: WARNING: cups/cups.h: present but cannot be compiled
configure: WARNING: cups/cups.h:     check for missing prerequisite headers?
configure: WARNING: cups/cups.h: see the Autoconf documentation
configure: WARNING: cups/cups.h:     section "Present But Cannot Be Compiled"
configure: WARNING: cups/cups.h: proceeding with the compiler's result
checking for cups/cups.h... no
configure: error: Could not find CUPS. Install libcups2-dev or cups-devel.
...
Running ./autogen from the folder /opt/ilbreoffice/ is successful without error.

Running the script as root on Ubuntut 16.04.3 LTS.
Any idea to solve this?

Thanks, Brakelmann

EDIT:
After fiddeling around I found the solution.
My lool user was not added correctly, something went wrong and the user was not in sudoers group.
So I deleted the user and the lool homedir.
After this the install script worked as expected.
